Sources/MapboxCoreNavigation/LegacyRouteController.swift

```diff
@@ -520,6 +520,7 @@ open class LegacyRouteController: NSObject, Router, InternalRouter, CLLocationMa
             guard forcedStepIndex < routeProgress.currentLeg.steps.count else { return }
             routeProgress.currentLegProgress.stepIndex = forcedStepIndex
         } else {
+            guard routeProgress.currentLegProgress.stepIndex < routeProgress.currentLeg.steps.endIndex - 1 else { return }
```

Comment:
Is `LegacyRouteController` really used anywhere at this point? As I can see it was deprecated in favor of `RouteController`, so not sure that such preventive check works.

Reply:
Thanks for the suggestion, I'll remove this part. I thought it would be better if we could keep it synced with the changes.